src/validateAtoms.ts

```diff
@@ -0,0 +1,88 @@
+import { atom, Getter, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+export type Validator = <Values extends Record<string, unknown>>(
+  values: Values,
+) => Promise<unknown>;
```

Comment:
Maybe this should be `unknown | Promise<unknown>` or `void | Promise<void>`. We support sync validator, right?

Reply:
that totally depends on the `loadable` API but I think the sync one should be working, i'll check in the evening

Reply:
done.
Though, there's an issue with using mixed typed atoms right now. The types for validator and labelled atoms needs to be fixed for that. 

On it next.